### PR TITLE
Adds Client Credentials Flow

### LINKED ISSUE #8: client_credentials grant flow
Described at https://github.com/stormpath/stormpath-framework-spec/blob/master/oauth2.md#client-credentials-grant-flow

### PR DESCRIPTION
**Pull request by edjiang**
Closes #8 

Also renamed methods
Also relaxed caching requirements so it’s only for responses with access tokens

See https://tools.ietf.org/html/rfc6749#section-5.1

> The authorization server MUST include the HTTP "Cache-Control" response header field [RFC2616] with a value of "no-store" in any response containing tokens, credentials, or other sensitive information, as well as the "Pragma" response header field [RFC2616] with a value of "no-cache".

### CODE REVIEW

File: src/test/groovy/com/stormpath/tck/oauth2/Oauth2IT.groovy [hunk 1]
```diff
@@ -239,4 +213,88 @@ class Oauth2IT extends AbstractIT {
...
+        given()
+            .param("grant_type", "client_credentials")
+            .auth()
+                .preemptive().basic(apiKeyId, apiKeySecret)
+            .contentType(ContentType.URLENC)
+        .when()
+            .post(OauthRoute)
+        .then()
+            .statusCode(200)
+            .contentType(ContentType.JSON)
```
Comment (nbarbettini) [inline src/test/groovy/com/stormpath/tck/oauth2/Oauth2IT.groovy:253]: Use `JsonResponseSpec.validAccessAndRefreshTokens` here?

Comment (edjiang) [inline src/test/groovy/com/stormpath/tck/oauth2/Oauth2IT.groovy:253]: That's why I wrote it, but actually `client_credentials` should not be outputting a refresh token. I can write a separate spec but it's not really necessary...

**Comment by nbarbettini** [comment]
PR looks good, pending above comment 👍

File: src/test/groovy/com/stormpath/tck/oauth2/Oauth2IT.groovy (hunk 1 above)
Comment (nbarbettini) [inline src/test/groovy/com/stormpath/tck/oauth2/Oauth2IT.groovy:253]: Ah, didn't catch that. Makes sense. LGTM

**Pull request merged**